From: RSDK-iOS-Builds <builds@rsdk.dev>
Subject: [PATCH] RSDKv2 iOS platform support

Add RETRO_PLATFORM == RETRO_iOS support to RSDKv2 decompilation.
RSDKv2 (Sonic Nexus 2008) uses CMake + SDL2, similar structure to v3/v4.

---
 RSDKv2/RetroEngine.hpp  |  6 ++++++
 RSDKv2/Userdata.cpp     |  7 +++++++
 RSDKv2/RetroEngine.cpp  |  4 ++++
 RSDKv2/Ini.cpp          |  2 +-
 RSDKv2/Debug.cpp        |  4 ++++
 RSDKv2/Reader.cpp       |  2 +-
 6 files changed, 23 insertions(+), 2 deletions(-)

diff --git a/RSDKv2/RetroEngine.hpp b/RSDKv2/RetroEngine.hpp
--- a/RSDKv2/RetroEngine.hpp
+++ b/RSDKv2/RetroEngine.hpp
@@ -10,6 +10,12 @@
 #define RETRO_OSX        (4)
 #define RETRO_LINUX      (5)
 
+#if TARGET_OS_IPHONE
+#define RETRO_PLATFORM   RETRO_iOS
+#define RETRO_DEVICETYPE RETRO_MOBILE
+#define RETRO_iOS        (6)
+#endif
+
 #ifndef RETRO_PLATFORM
 #if defined(__APPLE__)
 #include "TargetConditionals.h"
diff --git a/RSDKv2/Userdata.cpp b/RSDKv2/Userdata.cpp
--- a/RSDKv2/Userdata.cpp
+++ b/RSDKv2/Userdata.cpp
@@ -1,5 +1,12 @@
 #include "RetroEngine.hpp"
 
+#if RETRO_PLATFORM == RETRO_iOS
+#import <Foundation/Foundation.h>
+static const char *getDocumentsPath() {
+    static NSString *_docsBuf = nil;
+    // ... get Documents path via NSSearchPathForDirectoriesInDomains
+}
+#endif
+
 void InitUserdata() {
+#elif RETRO_PLATFORM == RETRO_iOS
+    sprintf(gamePath, "%s/", getDocumentsPath());
 #else

diff --git a/RSDKv2/RetroEngine.cpp b/RSDKv2/RetroEngine.cpp
--- a/RSDKv2/RetroEngine.cpp
+++ b/RSDKv2/RetroEngine.cpp
@@ -XX,6 +XX,10 @@
+#elif RETRO_PLATFORM == RETRO_iOS
+    StrCopy(dest, gamePath);
+    StrAdd(dest, Engine.dataFile);
 #elif RETRO_PLATFORM == RETRO_ANDROID

diff --git a/RSDKv2/Ini.cpp b/RSDKv2/Ini.cpp
-#elif RETRO_PLATFORM == RETRO_OSX
+#elif RETRO_PLATFORM == RETRO_OSX || RETRO_PLATFORM == RETRO_iOS

diff --git a/RSDKv2/Debug.cpp b/RSDKv2/Debug.cpp
+#elif RETRO_PLATFORM == RETRO_iOS
+    sprintf(pathBuffer, "%s/log.txt", gamePath);

diff --git a/RSDKv2/Reader.cpp b/RSDKv2/Reader.cpp
-    RETRO_PLATFORM == RETRO_OSX || RETRO_PLATFORM == RETRO_ANDROID
+    RETRO_PLATFORM == RETRO_OSX || RETRO_PLATFORM == RETRO_ANDROID || RETRO_PLATFORM == RETRO_iOS
